rust/connlib/clients/android/src/lib.rs

```diff
@@ -298,24 +307,35 @@ fn connect(
             source,
         }
     })?);
-    let portal_token = env
-        .get_string(&portal_token)
-        .map_err(|source| ConnectError::StringInvalid {
+    let portal_token = String::from(env.get_string(&portal_token).map_err(|source| {
+        ConnectError::StringInvalid {
             name: "portal_token",
             source,
-        })?
-        .into();
+        }
+    })?);
+    let device_id =
+        String::from(
+            env.get_string(&device_id)
+                .map_err(|source| ConnectError::StringInvalid {
+                    name: "device_id",
+                    source,
+                })?,
+        );
+    let log_dir =
+        String::from(
+            env.get_string(&log_dir)
+                .map_err(|source| ConnectError::StringInvalid {
+                    name: "log_dir",
+                    source,
+                })?,
+        );
     let callback_handler = CallbackHandler {
         vm: env.get_java_vm().map_err(ConnectError::GetJavaVmFailed)?,
         callback_handler,
     };
-    let device_id = env
-        .get_string(&device_id)
-        .map_err(|source| ConnectError::StringInvalid {
-            name: "device_id",
-            source,
-        })?
-        .into();
+
+    init_logging(log_dir.into());
```

Comment:
What if we return the guard from here and store it within `Session`?

Reply:
This is a good suggestion -- I'll take a stab at it this morning.